GW2EIEvtcParser/EIData/Buffs/BuffSimulators/BuffSimulationItems/BuffSimulationItem.cs

```diff
@@ -1,58 +1,58 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Diagnostics;
 using GW2EIEvtcParser.ParsedData;
 
-namespace GW2EIEvtcParser.EIData.BuffSimulators
+namespace GW2EIEvtcParser.EIData.BuffSimulators;
+
+/// <summary> A segment of time with type <see cref="double"/> with inclusive start and inclusive end. </summary>
+using Segment = GenericSegment<double>;
+
+internal abstract class BuffSimulationItem : AbstractSimulationItem
 {
-    internal abstract class BuffSimulationItem : AbstractSimulationItem
+    //NOTE(Rennorb): I changed the element to have start + end instead of start + duration to fix a bug. 
```

Comment:
What was the bug?

Reply:
To be fully honest don't quite remember. It had something to do with negative durations, but this section got chewed on quite a bit. Ive mainly kept it this way because the end time is the thing getting modified and read back for most operations, so it being represented by s+e makes it easier to reason about changes. 
As the comment states it should not matter much which way this is represented, but most operations are simpler in this version.

Reply:
I don't mind the change, end being the attribute matches better the useage indeed. As you have observed, it was initially following arc's start + duration logic. 
Without context, it seemed that you changed it to this in order to fix a bug that is currently present in master.

Reply:
yea i get the concern. I don't think it is something on master, I'm confident it was something i introduced and subsequently fixed. I will look into it again before this gets finished.